aten/src/ATen/core/ivalue_inl.h

```diff
@@ -521,14 +522,40 @@ std::vector<Elem> generic_to(
   return result;
 }
 
+template <typename T>
+c10::intrusive_ptr<T> IValue::toCustomClass() && {
+  static_assert(std::is_base_of<torch::CustomClassHolder, T>::value == true,
+    "toCustomClass requires that template parameter T must inherit "
+    "from torch::CustomClassHolder");
+  auto obj = toObject();
+  TORCH_CHECK(obj->slots().size() == 1,
+              "Tried to cast IValue to custom class but it did "
+              "not contain a custom class!");
+  // FIXME: this cast isn't actually checked!
```

Comment:
Options:
1. fix before land
2. name this unsafeToCustomClass until it checks this. Our users will find this method and try to call it.